webui/module/Application/src/Application/Controller/Plugin/SessionTimeoutPlugin.php

```diff
@@ -26,42 +26,43 @@
 namespace Application\Controller\Plugin;
 
 use Zend\Mvc\Controller\Plugin\AbstractPlugin;
+use Zend\Session\Container;
 
 class SessionTimeoutPlugin extends AbstractPlugin
 {
+   protected $session = null;
 
    public function timeout()
    {
       $configuration = $this->getController()->getServiceLocator()->get('config');
       $timeout = $configuration['configuration']['session']['timeout'];
 
       if($timeout === 0) {
-         return true;
-      }
-      else {
-         if($_SESSION['bareos']['idletime'] + $timeout > time()) {
-            $_SESSION['bareos']['idletime'] = time();
-            return true;
-         }
-         else {
-            session_destroy();
+         return false;
+      } else {
+         if(($this->session->offsetGet('idletime') + $timeout) > time()) {
+            $this->session->offsetSet('idletime', time());
             return false;
+         } else {
+            $this->session->getManager()->destroy();
+            return true;
          }
       }
    }
 
    public function isValid()
    {
-      if($_SESSION['bareos']['authenticated']) {
+      $this->session = new Container('bareos');
+
+      if($this->session->offsetGet('authenticated')) {
          if($this->timeout()) {
-            return true;
-         }
-         else {
             return false;
+         } else {
+            return true;
          }
```

Comment:
this could be replaced with a more readable (and shorter) `return !$this->timeout()`